tests/system/libraries/NvdaLib.py

```diff
@@ -111,6 +114,7 @@ def _createTestIdFileName(name):
 		suiteName = builtIn.get_variable_value("${SUITE NAME}")
 		testName = builtIn.get_variable_value("${TEST NAME}")
 		outputFileName = f"{suiteName}-{testName}-{name}".replace(" ", "_")
+		outputFileName = re.sub(RE_INVALID_FILENAME_CHRS, "", outputFileName)

```

Comment:
Perhaps you could use urllib.parse.quote? That would convert any non alphanumeric characters in to %xx (E.g. colon would be %3a).
This would continue to keep them unique, noting that percent itself would be converted to %25.